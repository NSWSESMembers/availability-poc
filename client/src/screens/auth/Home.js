import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { getCodePushHash } from '../../utils';
import DeviceInfo from '../../selectors/deviceInfo';
import Home from './components/Home';

class HomeScreen extends Component {
  static navigationOptions = () => ({
    header: null,
  });

  state = {
    codePushHash: '-',
    version: 'N/A',
  }

  componentWillMount() {
    getCodePushHash().then((value) => {
      this.setState({
        codePushHash: value,
      });
    });
    this.setState({
      version: DeviceInfo.getVersionString(),
    });
  }

  goToSignIn = () => {
    this.props.navigation.navigate('SignIn');
  };

  goToRegister = () => {
    this.props.navigation.navigate('SignUp');
  };

  render() {
    const { version, codePushHash } = this.state;
    return (
      <Home
        version={version}
        codePushHash={codePushHash}
        onPressSignIn={this.goToSignIn}
        onPressRegister={this.goToRegister}
      />
    );
  }
}

HomeScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
  }),
};

export default HomeScreen;
