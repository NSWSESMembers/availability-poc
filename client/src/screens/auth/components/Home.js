import React from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';

import Icon from 'react-native-vector-icons/FontAwesome';
import { Button } from '../../../components/Button';
import { Center, Container, Footer, Half, Holder } from '../../../components/Container';

const Home = ({ version, codePushHash, onPressSignIn, onPressOAuth, onPressRegister }) => (
  <Container isAlt>
    <Center>
      <Icon name="bullhorn" size={90} color="#FFF" />
      <Text style={{ fontSize: 45, color: '#FFF' }}>Callout</Text>
      <Text style={{ fontSize: 15, color: '#FFF' }}>
      Version {version} Revision {codePushHash}
      </Text>
    </Center>
    <Footer>
      <Half>
        <Holder>
          <Button text="OAuth" onPress={onPressOAuth} />
        </Holder>
      </Half>
      <Half>
        <Holder>
          <Button text="Sign In" onPress={onPressSignIn} />
        </Holder>
      </Half>
      <Half>
        <Holder>
          <Button text="Register" onPress={onPressRegister} type="secondary" />
        </Holder>
      </Half>
    </Footer>
  </Container>
);
Home.propTypes = {
  version: PropTypes.string,
  codePushHash: PropTypes.string,
  onPressSignIn: PropTypes.func,
  onPressOAuth: PropTypes.func,
  onPressRegister: PropTypes.func,
};

export default Home;
