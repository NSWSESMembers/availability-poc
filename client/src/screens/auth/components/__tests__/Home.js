import React from 'react';
import TestRenderer from 'react-test-renderer';

import Home from '../Home';

test('renders correctly', () => {
  const noop = jest.fn();
  const tree1 = TestRenderer.create(
    <Home
      version="version"
      codePushHash="codePushHash"
      onPressSignIn={noop}
      onPressRegister={noop}
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();
});
