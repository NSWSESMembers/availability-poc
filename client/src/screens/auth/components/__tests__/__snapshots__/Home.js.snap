// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#1e3c78",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 90,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï‚¡
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "fontSize": 45,
        }
      }
    >
      Callout
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "fontSize": 15,
        }
      }
    >
      Version 
      version
       Revision 
      codePushHash
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "height": 75,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "paddingLeft": 20,
              "paddingRight": 20,
              "width": "100%",
            },
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#f48603",
              "opacity": 1,
              "padding": 10,
              "width": "100%",
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            Sign In
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "paddingLeft": 20,
              "paddingRight": 20,
              "width": "100%",
            },
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#1c87c9",
              "opacity": 1,
              "padding": 10,
              "width": "100%",
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            Register
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
